sagan-common/src/main/java/sagan/projects/ProjectRelease.java

```diff
@@ -183,44 +187,75 @@ public ProjectRelease createWithVersionPattern() {
         return release;
     }
 
+    private static final Map<String, Integer> SPECIAL_MEANINGS =
+            ImmutableMap.of("dev", -1, "rc", 1, "release", 2, "final", 3);
+
+    /**
+     * Adapted from Gradle's StaticVersionComparator
+     */
     @Override
     public int compareTo(ProjectRelease other) {
-        if (this.getVersion() == null || other.getVersion() == null) {
-            return 0;
-        }
-
-        Pattern pattern = Pattern.compile("([0-9]+)");
+        if(other == null) return -1;
```

Comment:
To match existing coding style, put spaces between reserved words and the clause.

```
if (other == null)....
```